### DISCUSSION TITLE
side effects and meta props for field

### DISCUSSION THREAD

**Discussion by ha1fstack** [TanStack/form#709] [Ideas] [answered]
## 1. add side effects prop to field

It's a very common pattern to trigger side effects when field event triggers.
For example, I might want to reset other field when some field's `onChange` event triggers.
It's not impossible to implement in current version but it would be great if it was exposed to `Field`'s prop directly.

For workaround, I tried to add side effect to `validators` but validator always triggers on onSubmit so it caused unwanted behavior.

```tsx
<form.Field
  name="foo"
  effects={{
    onChange: ({ fieldApi }) => fieldApi.form.setFieldValue('someOtherField', undefined),
    // same fields with validators
  }}
>
```

## 2. add meta prop to field

Another common pattern is when you want to inject additional meta context to field from form status.
It's possible with `form.Subscribe` but it would be much convenient as a prop.

```tsx
<form.Field
  name="foo"
  meta={({ fieldApi }) => ({
    disabled: fieldApi.form.getFieldValue('someOtherField') == null,
    ...
  })}
  // or maybe expose values directly like form.Subscribe
  meta={({ values }) => ({
    disabled: values.someOtherField == null,
    ...
  })}
>
  {(field, meta) => <Comp
    disabled={meta.disabled}
    ... 
```

**Comment by crutchcorn** [comment, marked as answer]
These are actually _really_ impressive API ideas. Can you make GH issues for each of them and we can discuss each with their own merits?

**Reply by ha1fstack** [reply]
For sure, thanks! I might also create PR draft for this feature 😀
